### PR TITLE
Compile tests 

### PR DESCRIPTION
**Pull request by icristescu**

### CODE REVIEW

File: test/irmin-pack/common.ml [hunk 1]
```diff
@@ -95,60 +104,84 @@ struct
...
         ~flush_callback:(fun () -> !f ())
         ~log_size ~fresh name
     in
-    let indexing_strategy = Irmin_pack.Indexing_strategy.always in
-    let dict =
-      let path = Irmin_pack.Layout.V1_and_v2.dict ~root:name in
-      Irmin_pack_unix.Dict.v ~fresh ~readonly path
-    in
-    let io =
+    let dict = Dict.v ~capacity fm in
```
Comment (Ngoguey42) [inline test/irmin-pack/common.ml:162]: The file manager instantiates an index. Can we use it instead?

File: test/irmin-pack/common.mli [hunk 2]
```diff
@@ -73,27 +73,24 @@ end) : sig
...
   type t = {
-    lru_size : int;
     name : string;
+    fm : File_manager.t;
     index : Index.t;
     pack : read Pack.t;
-    dict : Irmin_pack_unix.Dict.t;
-    io : Irmin_pack_unix.Io_legacy.Unix.t;
+    dict : Dict.t;
+    io_v1_v2 : Irmin_pack_unix.Io_legacy.Unix.t;
```
Comment (Ngoguey42) [inline test/irmin-pack/common.mli:87]: `io_v1_v2` seem to only be opened and then closed. Can this be removed?

**Comment by icristescu** [comment]
All tests compile now, so this is ready.

**Review by Ngoguey42 — APPROVED** [review]
All these `get_ok` are annoying. Let's merge as is and replace them later when the file manager implements a clean way of raising exceptions

**Pull request merged**